### ISSUE TITLE
Some dailies and weeklies don't have names in histories

### ISSUE THREAD

**Issue by xparadoxical** [Cvolton/betterinfo-geode#5]

**Comment by Cvolton** [comment]
does this happen after restarting the game? usually this would mean the mod hasn't had time to load them yet, as they have to be fetched separately from the server (and then cached). Showing "Unknown" instead of the proper name would then be intended behavior, unless this sticks for a longer period of time

(this also applies to the latest daily/weekly if they were first downloaded during this game session, since this cache fetching happens on startup. This one could be fixed quite easily but I consider that more of a missing feature than a bug)

**Comment by Cvolton** [comment]
Not sure if this is what you meant but b4ce5196900c6767acc3cb0d6cdf7a9208a166ce addresses daily levels downloaded in the current game session showing up as "Unknown"

**Comment by xparadoxical** [comment]
> you've played the level normally and then deleted it from saved

Oh, I've done that a lot, if you mean saved levels and not history. If I complete a daily, then I claim the reward, complete the non-daily version too and delete it from saved levels.

**Comment by Cvolton** [comment]
I've just realized that there is a specific condition where the name of the current daily/weekly can change to "Unknown" - it appears that this would sometimes happen upon opening the daily level history page during the game session where a daily was first downloaded.

This should not be happening for any future daily/weekly levels as of the latest update, but I've added additional preventional measures in 3d601a20b914c6acd3b0d906477af17d5c81be7b to prevent this issue from happening at all in the future

As for the existing level where the name already got overwritten, I don't think there's a way to fix that without editing your save file. The easier way out would be just waiting until the next Weekly. The bug affects the level name, coin count (it's always set to 3 for affected levels) and it verifies coins (which I don't believe there has been a daily level with unverified coins anyways, so this isn't really a state change). The coin count inside the actual level obviously stays the same, as the level string isn't touched.

Do note that a permanent after effect of this would be the level name being stuck as "Unknown" in the weekly history page. To get around that I recommend modifying or outright deleting the cache.json file after switching to a different weekly.
